src/js/actions/documents.js

```diff
@@ -1003,6 +1004,21 @@ define(function (require, exports) {
         }.bind(this);
         descriptor.addListener("open", _openHandler);
 
+        _closeHandler = function (event) {
+            if (typeof event.documentID === "number") {
```

Comment:
We are calling an action, after filtering out some of the events. And the only reason we're filtering out events is due to buggy behavior in PS. I didn't want to implement an action around that behavior.